src/Actions/ResizeImages.php

```diff
@@ -5,26 +5,32 @@
 use JustBetter\ImageOptimize\Jobs\ResizeImageJob;
 use Statamic\Assets\Asset;
 use Statamic\Assets\AssetCollection;
+use Symfony\Component\Console\Helper\ProgressBar;
 
 class ResizeImages
 {
     public int $chunkSize = 200;
 
     public function __construct(
-        AssetCollection $assetCollection
+        AssetCollection $assetCollection,
+        public ?ProgressBar $progressBar = null
     ) {
         $this->resizeImages($assetCollection);
     }
 
     public function resizeImages(AssetCollection $assetCollection): void
     {
-        $assetCollection->chunk($this->chunkSize)->each(function (AssetCollection $assets) {
-            $assets->each(function (Asset $asset) {
-                if($asset->isImage()) {
-                    $asset->hydrate();
+        $assetCollection->lazy()->each(function (Asset $asset) {
+            if($asset->isImage()) {
+                $asset->hydrate();
+
+                if ($this->progressBar) {
+                    ResizeImageJob::dispatchSync($asset);
+                    $this->progressBar->advance();
+                } else {
                     ResizeImageJob::dispatch($asset);
```

Comment:
It would be nice to implement batching for this, loop and check over https://laravel.com/docs/9.x/queues#inspecting-batches to see how many jobs have been finished in order to update the progress bar until it's finished.

This way we maintain concurrency while adding the progress bar